llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -729,12 +731,14 @@ X86TargetLowering::X86TargetLowering(const X86TargetMachine &TM,
     setOperationAction(ISD::FSIN   , MVT::f32, Expand);
     setOperationAction(ISD::FCOS   , MVT::f32, Expand);
     setOperationAction(ISD::FSINCOS, MVT::f32, Expand);
+    setOperationAction(ISD::FTAN, MVT::f32, Expand);
```

Comment:
I think this should be unneeded due to TargetLoweringBase.cpp defaulting to Expand.

Reply:
A few questions. Firs if expand is default then why do we have so many Expands in this file? For example, in this block there is a sin\cos expand. 

Outside of the two cases you called out here there are other expands added in this file for tan.  Are you saying we should only be calling `setOperationAction` if we are doing `Action` or `LibCall`?

Reply:
sin/cos default to `Legal` because there are no calls to `setOperationAction` for them in TargetLoweringBase.cpp. The underlying array that `setOperationAction` writes to defaults to `Legal` in the constructor.

This patch modifies code in TargetLoweringBase.cpp to Expand ISD::TAN for f32/f64/f128. That makes any setOperationAction to `Expand` for ISD::TAN for f32/f64/f128 in X86ISelLowering.cpp redundant. Other types like f16, f80 or vectors still need to be done explicitly in X86ISelLowering.cpp.

```
  setOperationAction({ISD::FCBRT, ISD::FLOG, ISD::FLOG2, ISD::FLOG10, ISD::FEXP,
                      ISD::FEXP2, ISD::FEXP10, ISD::FFLOOR, ISD::FNEARBYINT,
                      ISD::FCEIL, ISD::FRINT, ISD::FTRUNC, ISD::LROUND,
                      ISD::LLROUND, ISD::LRINT, ISD::LLRINT, ISD::FROUNDEVEN,
                      ISD::FTAN},
                     {MVT::f32, MVT::f64, MVT::f128}, Expand);
```

Reply:
Thank you for the detailed answer. I noticed two other f32\f64 tan cases so removed those as well.